experimental/LieAlgebras/src/LieAlgebraIdeal.jl

```diff
@@ -99,21 +99,21 @@ function (I::LieAlgebraIdeal)()
 end
 
 @doc raw"""
-    (I::LieAlgebraIdeal{C})(v::Vector{Int}) -> LieAlgebraElem{C}
+    (I::LieAlgebraIdeal{C})(v::AbstractVector{Int}) -> LieAlgebraElem{C}
 
 Return the element of `I` with coefficient vector `v`.
 Fail, if `Int` cannot be coerced into the base ring of `I`.
 """
-function (I::LieAlgebraIdeal)(v::Vector{Int})
+function (I::LieAlgebraIdeal)(v::AbstractVector{Int})
   return I(coefficient_ring(I).(v))
 end
 
 @doc raw"""
-    (I::LieAlgebraIdeal{C})(v::Vector{C}) -> LieAlgebraElem{C}
+    (I::LieAlgebraIdeal{C})(v::AbstractVector{C}) -> LieAlgebraElem{C}
 
 Return the element of `I` with coefficient vector `v`.
 """
-function (I::LieAlgebraIdeal{C})(v::Vector{C}) where {C<:FieldElem}
+function (I::LieAlgebraIdeal{C})(v::AbstractVector{C}) where {C<:FieldElem}
```

Comment:
At first I was concerned that this change might lead to funky behaviour if `v` does not use 1-based indexing. But then I noticed the call `matrix(coefficient_ring(I), 1, length(v), v)` below, so I think that will throw a suitable error in that case. Fine by me then.

Reply:
This gets caught by https://github.com/Nemocas/AbstractAlgebra.jl/pull/1864/files#diff-77fc321e067710edc4d1df4198d7b3c983ee423acee3295a333ac68533656322R1076, so it should be fine